sklearn/cluster/tests/test_k_means.py

```diff
@@ -68,8 +70,70 @@ def test_kmeans_results(representation, algo, dtype):
     assert kmeans.n_iter_ == expected_n_iter
 
 
+@pytest.mark.parametrize("array_constr",
+                         [np.array, sp.csr_matrix],
+                         ids=['dense', 'sparse'])
+@pytest.mark.parametrize("algo", ['full', 'elkan'])
+def test_relocated_clusters(array_constr, algo):
```

Comment:
It's not a fix. I just added a test because the way empty clusters are relocated is a bit subtle and I've had a hard time reproducing the exact same scheme in the new implementation, so I added this test to make sure I was doing the right thing.